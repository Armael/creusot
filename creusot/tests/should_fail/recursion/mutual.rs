#![allow(unused)]
extern crate creusot_contracts;
use creusot_contracts::*;

mod m {
    pub fn f1() {
        super::f2();
    }
}

fn f2() {
    f3();
}

fn f3() {
    m::f1();
}
