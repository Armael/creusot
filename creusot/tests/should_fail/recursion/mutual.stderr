error: Mutually recursive functions: when calling m::f1...
  --> mutual.rs:6:5
   |
6  |     pub fn f1() {
   |     ^^^^^^^^^^^
   |
note: then m::f1 calls f2...
  --> mutual.rs:7:9
   |
7  |         super::f2();
   |         ^^^^^^^^^^^
note: then f2 calls f3...
  --> mutual.rs:12:5
   |
12 |     f3();
   |     ^^^^
note: finally f3 calls m::f1.
  --> mutual.rs:16:5
   |
16 |     m::f1();
   |     ^^^^^^^

error: aborting due to 1 previous error

